src/api/procedures/modifyAllowance.ts

```diff
@@ -0,0 +1,144 @@
+import BigNumber from 'bignumber.js';
+import { TxTags } from 'polymesh-types/types';
+
+import { PolymeshError, Procedure, Subsidy } from '~/internal';
+import { AllowanceOperation, ErrorCode } from '~/types';
+import { ProcedureAuthorization } from '~/types/internal';
+import { bigNumberToBalance, stringToAccountId } from '~/utils/conversion';
+
+interface IncreaseAllowanceParams {
+  /**
+   * amount of POLYX to be increased
+   */
+  allowance: BigNumber;
+  operation: AllowanceOperation.Increase;
+}
+
+interface DecreaseAllowanceParams {
+  /**
+   * amount of POLYX to be decreased
+   */
+  allowance: BigNumber;
+  operation: AllowanceOperation.Decrease;
+}
+
+interface SetAllowanceParams {
+  /**
+   * amount of POLYX to be set
+   */
+  allowance: BigNumber;
+  operation: AllowanceOperation.Set;
+}
+
+export type ModifyAllowanceParams = (
+  | IncreaseAllowanceParams
+  | DecreaseAllowanceParams
+  | SetAllowanceParams
+) & {
+  subsidy: Subsidy;
+};
```

Comment:
Since you're using `ModifyAllowanceParams` in the actual endpoints, the user won't see the separate doc comments for each case. I suggest exporting each interface and using those in the endpoints